source/module_hamilt_general/hamilt.h

```diff
@@ -33,7 +33,7 @@ class Hamilt
     int non_first_scf=0;
 
     // first node operator, add operations from each operators
-    Operator<std::complex<FPTYPE>, Device>* ops = nullptr;
+    Operator<T, Device>* ops = nullptr;
```

Comment:
When T=double, ops and opds will be same

Reply:
Sure. the next step I want to do is to unify opsd and ops.
I think there should be only one `Operator` instance in class `Hamilt`.

Reply:
T can not be double right now(This requires a lot of subsequent code changes, including updating the related classes and the underlying multi-device ops.) However, if we support T taking double in the future, can the opds be removed? Is it necessary to have both ops and opds in the basic class at the same time?

Reply:
> Sure. The next step I want to do is to unify opsd into ops. I think there should be only one `Operator` instance in class `Hamilt`.

Okay, I think so.